### PR TITLE
GitHub Action to lint Python code

### PR DESCRIPTION
**Pull request by cclauss**
Test results: https://github.com/cclauss/yulewalker/actions

### CODE REVIEW

**Comment by mmxgn** [comment]
I like this, 

@iver56  what do you say?

**Comment by iver56** [comment]
LGTM

File: .github/workflows/lint_python.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - run: black --check . || true
+      - run: codespell --ignore-words-list="fo,therefor"
+      - run: flake8 --ignore=E203,F401,F403 --max-complexity=10
+                    --max-line-length=125 --show-source --statistics .
+      - run: isort --check-only --profile black . || true
+      - run: pip install -r requirements.txt || pip install --editable . || pip install . || true
+      - run: mkdir --parents --verbose .mypy_cache
+      - run: mypy --ignore-missing-imports --install-types --non-interactive . || true
+      - run: pytest --doctest-modules . || true
+      - run: shopt -s globstar && pyupgrade --py37-plus **/*.py || true
```
Comment (mmxgn) [inline .github/workflows/lint_python.yml:24]: What does `shopt` do here?

**Review by mmxgn — APPROVED** [review]

File: .github/workflows/lint_python.yml (hunk 1 above)
Comment (cclauss) [reply .github/workflows/lint_python.yml:24]: `shopt -s globstar` allow bash to find all Python files recursively using `**/*.py` globbing.

This capability is on by default in `zsh`.

Comment (mmxgn) [reply .github/workflows/lint_python.yml:24]: Ah I see, thanks for the explanation. I learned something new :)

**Pull request merged**